drivers/storage/scaleio/storage/scaleio_storage.go

```diff
@@ -365,13 +366,21 @@ func (d *driver) VolumeCreate(ctx types.Context, volumeName string,
 	if opts.Type != nil {
 		volume.Type = *opts.Type
 	}
-	if opts.Size != nil {
-		volume.Size = *opts.Size
-	}
 	if opts.IOPS != nil {
 		volume.IOPS = *opts.IOPS
 	}
 
+	if opts.Size == nil {
+		volume.Size = int64(minSizeGiB)
+	}
+
+	fields["size"] = *opts.Size
```

Comment:
Hi @codenrhoden,

This will result in a runtime panic if `opts.Size` is `nil` since you cannot take the address of a `nil` value. Your pattern in the other sources accounts for this:

```go
if opts.Size == nil {
	opts.Size = &minSizeGiB
}
```